sys/net/gnrc/network_layer/ipv6/nib/_nib-slaac.c

```diff
@@ -30,8 +36,20 @@
 #if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_6LN) || IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_SLAAC)
 static char addr_str[IPV6_ADDR_MAX_STR_LEN];
 
+#if !IS_ACTIVE(CONFIG_GNRC_IPV6_STABLE_PRIVACY)
 void _auto_configure_addr(gnrc_netif_t *netif, const ipv6_addr_t *pfx,
                           uint8_t pfx_len)
+#else
+inline void _auto_configure_addr(gnrc_netif_t *netif, const ipv6_addr_t *pfx,
```

Comment:
Having this either-declared-or-inline is quite confusing, especially being split across two files; do you see options to simplify that?

Reply:
This either-declared-or-inline is to achieve a logic for method overloading.
If it's confusing, one could simply remove the `inline` qualifier 😄 (at slight cost of performance)

(Additionally, to revert this slight performance penalty,
one could move the logic for the method overload to each caller.
Like some existing code does (method has one caller only): https://github.com/RIOT-OS/RIOT/blob/32d1cd93a169335e5629f63d55ccc11406009681/sys/net/gnrc/network_layer/ipv6/nib/nib.c#L808-L816
With a definition like:
https://github.com/RIOT-OS/RIOT/blob/32d1cd93a169335e5629f63d55ccc11406009681/sys/net/gnrc/network_layer/ipv6/nib/nib.c#L1610-L1617

This would affect 3 existing callers. Instead of having it in one place to cover all callers at once, this puts the burden on the caller however, which may cause incomplete feature coverage if overlooked when adding a new caller.)

Reply:
I don't have the full picture, but if as I understand this should only affect callers in this compilation unit, would it work (i.e. maintain all the relevant properties) to have something like this?

```c
static inline void _our_auto_configure_addr(gnrc_netif_t *netif, const ipv6_addr_t *pfx,
                          uint8_t pfx_len)
{
#if !IS_ACTIVE(CONFIG_GNRC_IPV6_STABLE_PRIVACY)
    _auto_configure_addr(netif, pfx, len);
#else
    _auto_configure_addr_with_dad_ctr(netif, pfx, pfx_len, 0);
#endif
}
```